### PR TITLE
Imagenet adv generator

### LINKED ISSUE #241: Make imagenet_adversarial dataset return an ArmoryDataGenerator

### PR DESCRIPTION
**Pull request by seanpmorgan**
Closes #241

### CODE REVIEW

File: armory/baseline_models/keras/keras_resnet50.py [hunk 1]
```diff
@@ -1,7 +1,29 @@
...
+    output = []
+    for i in range(x.shape[0]):
+        im_raw = image.array_to_img(x[i])
+        im = image.img_to_array(im_raw.resize(shape))
+        output.append(im)
+    output = preprocess_input_resnet50(np.array(output))
+    return output
 
 
 def get_art_model(model_kwargs, wrapper_kwargs):
```
Comment (seanpmorgan) [inline armory/baseline_models/keras/keras_resnet50.py]: What should we set the clip_values to here? There is a different upper-bound for each channel.

Comment (davidslater) [reply armory/baseline_models/keras/keras_resnet50.py]: It's really hard for me to say. 
https://github.com/keras-team/keras-applications/blob/master/keras_applications/imagenet_utils.py
^ has a lot of different processing paths.

It depends on what the original model was trained on. They are zero-centered on ImageNet mean. Are they also scaled? Ideally, we would take the largest inputs from the original domain for each channel ([0, 255], I think?), then apply the preprocessing to get the range for each channel. If the per-channel domains are different, then we'll need to pass in a tuple of np.arrays to the clip_values kwarg.

Comment (seanpmorgan) [reply armory/baseline_models/keras/keras_resnet50.py]: So I took the preprocessing functions for each model and pulled them out of keras.applications to limit confusion. This particular model uses:

I didn't know clip_values could be set per dimension, [but it looks as though they can](https://github.com/IBM/adversarial-robustness-toolbox/blob/master/art/classifiers/classifier.py#L47-L50). PTAL at the changes.

Comment (davidslater) [reply armory/baseline_models/keras/keras_resnet50.py]: I'll need to play around with the changes to ensure this is what ART expects, but this looks good.

Comment (davidslater) [reply armory/baseline_models/keras/keras_resnet50.py]: Yes, this looks good. 

ART uses `np.clip` under the hood, so it broadcasts the clip values.

**Pull request merged**